src/main/java/uk/co/onsdigital/discovery/metadata/api/dao/MetadataDaoImpl.java

```diff
@@ -61,4 +64,17 @@ public ConceptSystem findConceptSystemByDataSetAndConceptSystemName(String dataS
                 .findAny()
                 .orElseThrow(() -> new ConceptSystemNotFoundException("No such concept system in dataset: " + conceptSystemName));
     }
+
+    @Override
+    public GeographicAreaHierarchy findGeographyInDataSet(String dataSetId, String geographyId)
+            throws DataSetNotFoundException, GeographicHierarchyNotFoundException {
+
+        // FIXME: this is a provisional inefficient implementation for the current prototype.
+        // At some point the metadata database will be redesigned to not store data points, at which point this should be optimised.
+        final DimensionalDataSet dataSet = findDataSetById(dataSetId);
+        return dataSet.getReferencedGeographies()
+                .filter(geog -> Objects.equals(geog.getGeographicAreaHierarchy(), geographyId))
+                .findAny()
+                .orElseThrow(() -> new GeographicHierarchyNotFoundException(geographyId));
```

Comment:
As the comment mentions, this takes the naive approach of iterating through all data points in the dataset. At some point the loader needs to switch from loading the actual data points to just the metadata, at which point it should instead populate a summary table with this information directly.